src/gridfs.rs

```diff
@@ -0,0 +1,168 @@
+use core::task::{Context, Poll};
+use std::{
+    io::{self, Result},
+    marker::PhantomPinned,
+    pin::Pin,
+};
+
+use crate::{
+    concern::{ReadConcern, WriteConcern},
+    cursor::Cursor,
+    selection_criteria::ReadPreference,
+    Database,
+};
+
+use bson::{oid::ObjectId, DateTime, Document};
+use serde::Deserialize;
+use typed_builder::TypedBuilder;
+
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFSBucketOptions {
+    bucket_name: Option<String>,
+    chunk_size_bytes: Option<i32>,
+    write_concern: Option<WriteConcern>,
+    read_concern: Option<ReadConcern>,
+    read_preference: Option<ReadPreference>,
+}
+
+pub struct GridFSUploadOptions {
```

Comment:
We'll need to add the same annotations as are included on `GridFSBucketOptions` here, and ditto for all other options structs.